libgnucash/engine/Split.c

```diff
@@ -1272,6 +1272,12 @@ xaccSplitGetBalance (const Split *s)
     return s ? s->balance : gnc_numeric_zero();
 }
 
```

Comment:
You also need to initialize noclosing_balance in `gnc_split_init` and `xaccSplitReinit`, copy it in `xaccSplitCloneNoKVP`, report it in `xaccSplitDump`, and test it in `xaccSplitEqual`.